packages/@react-native-windows/cli/src/generator-windows/index.ts

```diff
@@ -515,53 +516,46 @@ function toCppNamespace(namespace: string) {
   return namespace.replace(/\./g, '::');
 }
 
-export function installDependencies(options: {verbose: boolean}) {
-  const cwd = process.cwd();
-
-  // Extract react-native peer dependency version
-  const rnwPackageJsonPath = require.resolve(
-    'react-native-windows/package.json',
-    {paths: [process.cwd()]},
-  );
-  const rnwPackageJson = JSON.parse(
-    fs.readFileSync(rnwPackageJsonPath, {encoding: 'UTF8'}),
-  );
-  let rnPeerDependency = rnwPackageJson.peerDependencies['react-native'];
-  const depDelim = ' || ';
-  const delimIndex = rnPeerDependency.indexOf(depDelim);
-  if (delimIndex !== -1) {
-    rnPeerDependency = rnPeerDependency.slice(0, delimIndex);
+export async function installScriptsAndDependencies(options: {
+  verbose: boolean;
+}) {
+  const projectPackage = await WritableNpmPackage.fromPath(process.cwd());
+  if (!projectPackage) {
+    throw new Error('The current directory is not the root of an npm package');
   }
 
-  const rnPackageJsonPath = require.resolve('react-native/package.json', {
-    paths: [process.cwd()],
+  await projectPackage.mergeProps({
+    scripts: {windows: 'react-native run-windows'},
   });
-  const rnPackageJson = JSON.parse(
-    fs.readFileSync(rnPackageJsonPath, {encoding: 'UTF8'}),
-  );
 
-  if (!semver.satisfies(rnPackageJson.version, rnPeerDependency)) {
+  const rnwPackage = await findPackage('react-native-windows');
+  if (!rnwPackage) {
+    throw new Error('Could not locate the package for react-native-windows');
+  }
+
+  const rnPackage = await findPackage('react-native');
+  if (!rnPackage) {
+    throw new Error('Could not locate the package for react-native');
+  }
+
+  const rnPeerDependency = rnwPackage.json.peerDependencies['react-native'];
+
+  if (
+    !semver.satisfies(rnPackage.json.version, rnPeerDependency) &&
+    projectPackage.json.dependencies?.['react-native']
+  ) {
     console.log(
       chalk.green('Installing a compatible version of react-native:'),
     );
     console.log(chalk.white(`    ${rnPeerDependency}`));
 
     // Patch package.json to have proper react-native version and install
-    const projectPackageJsonPath = path.join(cwd, 'package.json');
-    const projectPackageJson = JSON.parse(
-      fs.readFileSync(projectPackageJsonPath, {encoding: 'UTF8'}),
-    );
-    projectPackageJson.scripts.windows = 'react-native run-windows';
-    if (projectPackageJson.hasOwnProperty('dependencies')) {
-      projectPackageJson.dependencies['react-native'] = rnPeerDependency;
-    }
-    fs.writeFileSync(
-      projectPackageJsonPath,
-      JSON.stringify(projectPackageJson, null, 2),
-    );
+    await projectPackage.mergeProps({
+      dependencies: {'react-native': rnPeerDependency},
+    });
 
```

Comment:
nit: for completeness in the future: Should we check if users perhaps have one of the dependencies as `devDependencies` rather than `dependencies`?

Reply:
I have seen lib modules specify `react`, `react-native`, and `react-native-windows` as:

1. Only devDependencies
2. devDependencies + peerDependencies (not always matching)

I've yet to see one only specify in peerDependencies, but I imagine it's out there. They definitely never specify any of those as regular dependencies.

Does this mean this code will not bump the RN version for modules, since `projectPackage.json.dependencies?.['react-native']` will always evaluate to false?

I think this can be a separate bug, but I'd have to think through the implications. Modules can have very broad version ranges as peer/dev dependencies (like `lottie-react-native` requires `react-native` with peer `">=0.46"` and dev `"0.62.2"`). I think the correct answer for that specific module would be to bump the dev version. That would enable adding windows correctly to the module, and being able to work on it. Then when it gets consumed by an app, you're relying on the app's project getting updated with the real correct `react-native` dependency, which the modules consumes as a peer.

Reply:
I agree that knowing the right thing to do is tricky. Previous behavior was to always assign a `dependency` if the object is available, even in the community module case where we would want to bump `devDependency`. New behavior is to instead noop, which is probably better, but won't get everything into the right state either.

I think the whole version matching logic we have could be smarter. We have some different logic and different behavior split between `react-native-windows-init` and the generator as well which would be great to refactor and get testable.

FWIW, this is the code we use to upgrade dependencies of our whole repo https://github.com/microsoft/react-native-windows/blob/master/packages/%40rnw-scripts/integrate-rn/src/upgradeDependencies.ts Doing the right things here ends up being a super tricky problem.